### DISCUSSION TITLE
Getting resource from menu

### DISCUSSION THREAD

**Discussion by patjakubik** [Shopify/storefront-api-feedback#116] [Ideas & Requests]
I'm really glad that querying menu is now possible. However it has one big limitation compared to Liquid. For example, we would like to display image of linked category or a product, and this is not possible without making additional query, as Menu returns only `resourceId`, `title` and `url`.

It would be nice to have `resource` attribute, like this:
![CleanShot 2022-06-13 at 18 16 35@2x](https://user-images.githubusercontent.com/37963339/173398625-3c078574-9c9f-4c22-ae62-af4844cc7694.png)
:

**Comment by vixdug** [comment]
Hi @patryk-smc - Thanks for this feedback. It is something I've been thinking of as well, just to confirm is your use case to build a "mega menu" type of navigation?

**Reply by patjakubik** [reply]
Not really mega menu, just menu. And we also use menus to power pages that list things like featured collections.

**Comment by ghost** [comment]
This is also something we're looking for! Mainly for a mega menu (like https://tailwindui.com/components/ecommerce/components/store-navigation) where we would need to fetch images stored in collection metafields. This would improve performance a lot instead of having to fetch the menu, then go through results to gather the collections ids and then fetching the corresponding collections.

Thanks!

**Reply by gauravagarwal2704** [reply]
This is exactly what I need too ! @vixdug you think this can be available anytime soon ?

**Reply by bitforcesrl** [reply]
we've opened a similar thread here: https://github.com/Shopify/storefront-api-feedback/discussions/161

**Reply by gauravagarwal2704** [reply]
@johncraigcole looking for the solution over here. Any help is appreciated.

**Reply by bitforcesrl** [reply]
Hi @gauravagarwal2704 , @vixdug  is on this topic and she's from the API team, she could also give us some updates.
Thanks

**Comment by johncraigcole** [comment]
Hey Team 👋. We just pushed the functionality @patryk-smc proposed where you're able to query the resource of a menu item. Available for testing only in `unstable`, and is targeted for formal release in `2023-07`.

**Reply by patjakubik** [reply]
Awesome news! Thanks John
